cluster_management/src/main/java/com/pinterest/rocksplicator/Utils.java

```diff
@@ -99,6 +100,25 @@ public static ExternalView getHelixExternalViewWithFixedRetry(HelixAdmin admin,
     return null;
   }
 
+   /**
+   * Obtain the state map of a partition in the external view
+   * @param view Helix external view for a resource
+   * @param partitionName partition name to look up
+   * @return StateMap, which will be empty if the partition is not found, or there is no instance in the partition
+   */
+  public static Map<String, String> getStateMap(ExternalView view, String partitionName) {
+    if (partitionName == null || partitionName.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    Map<String, String> result = view.getStateMap(partitionName);
```

Comment:
Is it possible that the `view` is also Null? (will that Helix funct ever return Null?)

Reply:
yeah this could be `null` as well, e.g. when it failed to contact Helix.  I have previously added [retry](https://github.com/pinterest/rocksplicator/pull/598) on getting helix external view so it's less likely, but theoretically it can still happen. I've update the PR to check for `view` as well for additional safety.